Esri/arcgis-maps-sdk-swift-samples
# [Update] Add favorites to the Sample Viewer

## Description

This PR implements the ability to favorite samples in the Sample Viewer similar to the old iOS Sample Viewer. A "Favorites" category is was added, which allows the user to view a list of the samples that they favorited in the order that they were added. Samples in the list can be removed and rearranged by tapping the "Edit" button. Samples can be favorited/unfavorited by entering a sample and pressing the star button on the top toolbar. 

## Linked Issue(s)

- `swift/issues/4370`

## How To Test

- Add a sample to the "Favorites" list by launching a sample and tapping the star button on the top toolbar.
- Back out of the sample and enter the "Favorites" category to verify the sample shows up in the list.
- Remove a sample from the list by re-tapping the star button in the sample or by swiping right on a sample in the list.
- Tap the "Edit" to remove or rearrange samples in the list.
- Close and reopen Sample Viewer to verify the favorited samples remain in the list. 

## Screenshots

|Before|After|
|:-:|:-:|
|![Categories, Before](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/b0891aec-e7a6-4d23-a5db-bd79722204be)|![Categories, After](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/793fb3e5-50c4-48db-98ad-cbc41e7eb0ed)|
|![Sample, Before](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/c9fe733e-b3bf-49a6-96e8-47b687ed9557)|![Sample, After](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/99304db6-49a8-4b16-b514-ba290713959a)|
||![Favorites, After](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/28b95a12-ea14-473f-a1d7-2202b8f1145e)|
